### PR TITLE
[Ready] Cargo MK III - Glassware Addtion

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request
Needs #9912 to be merged first
Lowers the cost of cap pen
Fixes a few crates being unattainable as well as fixing a few broken crates.
Lowers how much booze bottler takes by upto half
Adds Glassworking, a simple mech-like construction that allows cargo to be able to make nice glassware/money for doing some work.
Adds glassworking pack to sci that allows people to make the new glass ware

## Why It's Good For The Game

Fixes and Tweaking for ideas as well as slowly rolling out the next idea for starter fixes and such
Makes cargo have a new means of making money using a mat that tends to be over looked and only used for windows/prolathens!

## Changelog
:cl:
add: more glassware
add: Few new packs and glassmaker kit!
tweak: prices/chems of crates/autobottler
fix: some crates ungettable as well as some broken crates
/:cl:

### CODE REVIEW

File: code/modules/cargo/bounties/silly.dm [hunk 1]
```diff
@@ -51,4 +51,11 @@
 	name = "Strange Object"
 	description = "Nanotrasen has taken an interest in strange objects. Find one in maint, and ship it off to CentCom right away."
 	reward = 1000
-	wanted_types = list(/obj/item/relic)
\ No newline at end of file
+	wanted_types = list(/obj/item/relic)
+
+/datum/bounty/item/silly/toy_swords
+	name = "Toy Swords"
+	description = "Nanotrasen has partaken in a reenactment of a famous movie, but we only have lethal energy based swords. Send us some toys so we stop killing are actors."
```
Comment (Poojawa) [inline code/modules/cargo/bounties/silly.dm]:
```suggestion
	description = "Nanotrasen has secured a contract in a reenactment of a famous movie, but we only have lethal energy based swords. Send us some convincing replicas so we save on stuntmen casualty re-hiring."
```

File: code/modules/reagents/reagent_containers/glass.dm [hunk 2]
```diff
@@ -8,6 +8,14 @@
 	resistance_flags = ACID_PROOF
 	container_HP = 2
 
+/obj/item/reagent_containers/glass/slug
+	name = "glass slug"
+	desc = "Seemingly useless wasted sand and fuel mixed in one... Or maybe it can be refined into something more usefull?"
```
Comment (Poojawa) [inline code/modules/reagents/reagent_containers/glass.dm]:
```suggestion
	desc = "A small slug of solid glass, produced from a kiln."
```

File: code/game/machinery/kilm.dm [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			if(S.amount < 5)
+				qdel(S)
+			else
+				user.show_message("<span class='notice'>You need a at lest five sand piles to make anything of use.</span>", 1)
+
+/obj/machinery/kilm/attack_hand(mob/living/carbon/user)
+	. = ..()
+	if(.)
+		return
+	if(charges == 0)
```
Comment (Poojawa) [inline code/game/machinery/kilm.dm]: I feel like this could be made a switch statement.

It should also display how much glass is in the kiln "You estimate there's about [charges] worth of sand in the hopper"

There should technically be just 'molten glass slug' that it produces, and requires a set of tongs in crafting to fashion each one out, instead of it magically making one of these items. not doing the crafting just makes a glass slug. just take the code from microwaved donk pockets for that timer.

Comment (Trilbyspaceclone) [reply code/game/machinery/kilm.dm]: Good idea thank you

Comment (Trilbyspaceclone) [reply code/game/machinery/kilm.dm]: The kilm is for the moment being scrapped do to incompetents on my part

File: code/modules/cargo/packs/medical.dm [hunk 4]
```diff
@@ -194,6 +183,18 @@
...
+	desc = "Contains five synthetics blood packs for reintroducing blood to patients."
+	cost = 3000
+	contains = list(/obj/item/reagent_containers/blood/synthetics,
+					/obj/item/reagent_containers/blood/synthetics,
+					/obj/item/reagent_containers/blood/synthetics,
+					/obj/item/reagent_containers/blood/synthetics,
+					/obj/item/reagent_containers/blood/synthetics)
+	crate_name = "blood freezer"
+	crate_type = /obj/structure/closet/crate/freezer
+
```
Comment (Ghommie) [inline code/modules/cargo/packs/medical.dm]: I thought I did remove it. Damnit.

File: code/modules/cargo/packs/misc.dm [hunk 5]
```diff
@@ -212,12 +200,55 @@
...
+				/obj/item/reagent_containers/food/drinks/bottle/wine,
+				/obj/item/reagent_containers/food/drinks/bottle/wine,
+				/obj/item/stack/rods/ten,
+				/obj/item/stack/sheet/metal/twenty,
+				/obj/item/stack/sheet/mineral/wood/twenty,
+				/obj/item/storage/box/drinkingglasses,
+				/obj/item/storage/box/drinkingglasses)
+	crate_name = "romance crate"
+
+/datum/supply_pack/misc/cbtpack
```
Comment (Ghommie) [inline code/modules/cargo/packs/misc.dm]: Please remove this joke.

File: code/modules/cargo/packs/misc.dm [hunk 6]
```diff
@@ -212,12 +200,55 @@
 					/obj/item/circuitboard/computer/slot_machine)
 	crate_name = "casino crate"
 
+/datum/supply_pack/misc/candlelitdinner
```
Comment (Ghommie) [inline code/modules/cargo/packs/misc.dm]: Remove this one too, wood and carpet supply packs exist, as well as the booze-o-mat.

File: code/modules/cargo/packs/organic.dm [hunk 7]
```diff
@@ -249,6 +235,29 @@
 	crate_name = "wildcard food crate"
 	crate_type = /obj/structure/closet/crate/freezer
 
+/datum/supply_pack/organic/party
```
Comment (Ghommie) [inline code/modules/cargo/packs/organic.dm:233]: Remove, there is already one located in service.dm

File: code/modules/cargo/packs/service.dm [hunk 8]
```diff
@@ -230,10 +253,56 @@
...
+
+/datum/supply_pack/service/carpet2
+	name = "Premium Carpet Crate #2"
+	desc = "Plasteel floor tiles getting on your nerves? These stacks of extra soft carpet will tie any room together.  Contains red, and monochrome"
+	cost = 1000
+	contains = list(/obj/item/stack/tile/carpet/blackred/fifty,
+					/obj/item/stack/tile/carpet/blackred/fifty,
+					/obj/item/stack/tile/carpet/monochrome/fifty,
+					/obj/item/stack/tile/carpet/monochrome/fifty)
+	crate_name = "premium carpet crate #2"
```
Comment (Ghommie) [inline code/modules/cargo/packs/service.dm]: Remove this accidental revert and move the new premium carpet supply pack here instead (you can find it in packs/misc.dm).

File: code/modules/cargo/packs/service.dm [hunk 9]
```diff
@@ -230,10 +253,56 @@
...
+
+/datum/supply_pack/service/carpet_exotic
```
Comment (Ghommie) [inline code/modules/cargo/packs/service.dm]: You are leaving the duplicate untouched in misc.dm.

File: code/modules/crafting/glassware.dm [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+/obj/item/glasswork/glass_base/dish
+	name = "Glass Fodder Sheet"
+	desc = "A set of glass sheets set aside for glass working, this one is ideal for a small glass dish. Needs to be cut with some tools."
+	next_step = /obj/item/glasswork/glass_base/dish_part1
+
+/obj/item/glasswork/glass_base/dish/attackby(obj/item/I, mob/user, params)
+	..()
+	if(I.tool_behaviour == TOOL_GLASS_CUT)
+		new next_step(user.loc, 1)
+		qdel(src)
```
Comment (Ghommie) [inline code/modules/crafting/glassware.dm:53]: I see lot of this copypaste around, could you please move it to their parent types (and perhaps add an extra check to avoid bad things from happening should next_step be null) instead?
Heck, maybe make a new proc that can be overriden for certain peculiar steps such as the glass fodder sheet which also deletes the tool.

File: code/modules/crafting/glassware.dm [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+	desc = "Failer to code. Contact your local bug remover..."
+	icon = 'icons/obj/glassworks.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	force = 1
+	throw_speed = 1
+	throw_range = 3
+	tool_behaviour = null
+
+/obj/item/glasswork/glasskit
+	name = "Glass Working Tools"
```
Comment (Ghommie) [inline code/modules/crafting/glassware.dm]: They are not proper names, please lowercase them.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 12]
```diff
@@ -158,9 +156,54 @@
...
+	name = "glass dish"
+	desc = "A tiny glass dish. It can hold up to 3 units. Unable to withstand reagents of an extreme pH."
+	materials = list(MAT_GLASS=500)
+	icon_state = "glass_disk"
+	possible_transfer_amounts = list(0.1,0.5,0.75,1,2,3)
+	volume = 3
+
+/obj/item/reagent_containers/glass/beaker/glass_dish/Initialize()
+	beaker_weakness_bitflag &= ~PH_WEAK
+	. = ..()
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]: Just add `beaker_weakness_bitflag  = NONE` to the typepath vars and delete these lines.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 13]
```diff
@@ -158,9 +156,54 @@
...
+
+/obj/item/reagent_containers/glass/beaker/flask_large
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]: Tell me why the `flask_large`, `flaskspouty` and `flask_small` paths shouldn't be `flask/large`, `flask/spouty` and `flask` respectively.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Required changes, and you apparently did not resolve a few conflicts correctly.

File: code/modules/cargo/packs/medical.dm (hunk 4 above)
Comment (Trilbyspaceclone) [reply code/modules/cargo/packs/medical.dm]: Is this intended to be removed or not?

File: code/modules/reagents/reagent_containers/glass.dm (hunk 12 above)
Comment (Trilbyspaceclone) [reply code/modules/reagents/reagent_containers/glass.dm]: These lines are needed so that they beakers understand their volume on creation

File: code/modules/reagents/reagent_containers/glass.dm (hunk 13 above)
Comment (Trilbyspaceclone) [reply code/modules/reagents/reagent_containers/glass.dm]: Consistency

File: code/modules/cargo/packs/medical.dm (hunk 4 above)
Comment (Ghommie) [reply code/modules/cargo/packs/medical.dm]: The synthetic blood pack was merged with the standard bloock pack, as the former surpasses the latter by a noticeable margin in terms of efficiency, little balance issue.

File: code/modules/reagents/reagent_containers/glass.dm (hunk 12 above)
Comment (Ghommie) [reply code/modules/reagents/reagent_containers/glass.dm]: This is the first and only operation done on these beakers' `Initialize()` before the parent call,  how would it even differ, gameplay wise, from having the bitflag defined on the typepath? Please keep it consistent with all the other datums that override initial vars on their path instead of doing so on `Initialize()`.
Otherwise explain yourself, cause I can't even catch a glimpse of that comment.

File: code/modules/reagents/reagent_containers/glass.dm (hunk 13 above)
Comment (Ghommie) [reply code/modules/reagents/reagent_containers/glass.dm]: In layman terms, common sense tells me a large flask is still a flask, so why wouldn't it be a flask subtype?
Please explain me how could this little bloat of unintuitive copypasta be defined as "Consistency".

File: code/modules/crafting/glassware.dm (hunk 10 above)
Comment (Putnam3145) [reply code/modules/crafting/glassware.dm:53]: Funnily enough, I suggested this because of the whole "parent type proc is getting called before subtype proc" thing.

File: code/modules/reagents/reagent_containers/glass.dm (hunk 13 above)
Comment (Trilbyspaceclone) [reply code/modules/reagents/reagent_containers/glass.dm]: all the different beaker types are based on the glass beaker in this file.
Even things like metamatteral being not a subtype below large.
I cant fine a single beaker thats past the subtype of the top of the file beaker

Comment (Trilbyspaceclone) [reply code/modules/reagents/reagent_containers/glass.dm]: Well past bucket is water bottle and water bottle large

File: code/modules/reagents/reagent_containers/glass.dm (hunk 12 above)
Comment (Trilbyspaceclone) [reply code/modules/reagents/reagent_containers/glass.dm]: without that line when made they have a volume of 1
also they are weak to PH so the flag is needed.

File: code/modules/cargo/packs/medical.dm (hunk 4 above)
Comment (Trilbyspaceclone) [reply code/modules/cargo/packs/medical.dm]: Er do you wish for me to tweak it a bit to cost more or remove it?

File: code/modules/reagents/reagent_containers/glass.dm (hunk 13 above)
Comment (Trilbyspaceclone) [reply code/modules/reagents/reagent_containers/glass.dm]: Also having them subtypes does nothing really as all the vars are different.

File: code/modules/reagents/reagent_containers/glass.dm (hunk 12 above)
Comment (Ghommie) [reply code/modules/reagents/reagent_containers/glass.dm]: That's because you unawarely wrote down `volume = NONE` instead of `beaker_weakness_bitflag = NONE`. Please be more careful next time and do what I said.

File: code/modules/reagents/reagent_containers/glass.dm (hunk 13 above)
Comment (Ghommie) [reply code/modules/reagents/reagent_containers/glass.dm]: Sadly I'm having hard time understanding your flow of logic here. But I'll repeat myself.
the large flask and the flask with spout should be flask subtypes for better maintainability should all three share more similar charateristics in the future, starting with their lack of PH_WEAK beaker bitflag.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 14]
```diff
@@ -159,8 +157,49 @@
...
+
+/obj/item/reagent_containers/glass/beaker/flask_small
+	name = "small flask"
+	desc = "A small flask. It can hold up to 40 units. Unable to withstand reagents of an extreme pH."
+	materials = list(MAT_GLASS=1000)
+	icon_state = "flasksmall"
+	volume = 40
+
+/obj/item/reagent_containers/glass/beaker/flask_small/Initialize()
+	. = ..()
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]: Remove this.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 15]
```diff
@@ -159,8 +157,49 @@
...
+/obj/item/reagent_containers/glass/beaker/flaskspouty
+	name = "flask with spout"
+	desc = "A flask with a spout! It can hold up to 120 units. Unable to withstand reagents of an extreme pH."
+	materials = list(MAT_GLASS=2500)
+	icon_state = "flaskspouty"
+	possible_transfer_amounts = list(1,2,3,4,5,10,15,20,25,30,50,100,120)
+	volume = 120
+
+/obj/item/reagent_containers/glass/beaker/flaskspouty/Initialize()
+	. = ..()
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]: Idem.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 16]
```diff
@@ -159,8 +157,49 @@
...
+
+/obj/item/reagent_containers/glass/beaker/flask_small/flask_large
+	name = "large flask"
+	desc = "A large flask. It can hold up to 80 units. Unable to withstand reagents of an extreme pH."
+	materials = list(MAT_GLASS=2500)
+	icon_state = "flasklarge"
+	volume = 80
+
+/obj/item/reagent_containers/glass/beaker/flask_small/flask_large/Initialize()
+	. = ..()
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]: Same.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 17]
```diff
@@ -159,8 +157,49 @@
...
+/obj/item/reagent_containers/glass/beaker/glass_dish
+	name = "glass dish"
+	desc = "A tiny glass dish. It can hold up to 3 units. Unable to withstand reagents of an extreme pH."
+	materials = list(MAT_GLASS=500)
+	icon_state = "glass_disk"
+	possible_transfer_amounts = list(0.1,0.5,0.75,1,2,3)
+	volume = 3
+
+/obj/item/reagent_containers/glass/beaker/glass_dish/Initialize()
+	. = ..()
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]: Ditto.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/cargo/packs/science.dm [hunk 18]
```diff
@@ -82,6 +81,27 @@
 					/obj/item/integrated_electronics/wirer)
 	crate_name = "circuitry starter pack crate"
 
+/datum/supply_pack/science/glasswork
+	name = "GlassBlower Gear Crate"
```
Comment (Ghommie) [inline code/modules/cargo/packs/science.dm]:
```suggestion
	name = "Glass blower kit Crate"
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagent_containers/glass.dm [hunk 19]
```diff
@@ -159,8 +157,37 @@
...
+	volume = 3
+
+/obj/item/reagent_containers/glass/beaker/flask_small/flask_large
+	name = "large flask"
+	desc = "A large flask. It can hold up to 80 units. Unable to withstand reagents of an extreme pH."
+	materials = list(MAT_GLASS=2500)
+	icon_state = "flasklarge"
+	volume = 80
+
+/obj/item/reagent_containers/glass/beaker/flask_small
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]: Just rename the path to:
```
/obj/item/reagent_containers/glass/beaker/flask
```

File: code/modules/reagents/reagent_containers/glass.dm [hunk 20]
```diff
@@ -159,8 +157,37 @@
...
+
+/obj/item/reagent_containers/glass/beaker/glass_dish
+	name = "glass dish"
+	desc = "A tiny glass dish. It can hold up to 3 units. Unable to withstand reagents of an extreme pH."
+	materials = list(MAT_GLASS=500)
+	icon_state = "glass_disk"
+	possible_transfer_amounts = list(0.1,0.5,0.75,1,2,3)
+	volume = 3
+
+/obj/item/reagent_containers/glass/beaker/flask_small/flask_large
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]:
```
/obj/item/reagent_containers/glass/beaker/flask/large
```

File: code/modules/reagents/reagent_containers/glass.dm [hunk 21]
```diff
@@ -159,8 +157,37 @@
...
+	volume = 80
+
+/obj/item/reagent_containers/glass/beaker/flask_small
+	name = "small flask"
+	desc = "A small flask. It can hold up to 40 units. Unable to withstand reagents of an extreme pH."
+	materials = list(MAT_GLASS=1000)
+	icon_state = "flasksmall"
+	volume = 40
+
+/obj/item/reagent_containers/glass/beaker/flaskspouty
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]:
```
/obj/item/reagent_containers/glass/beaker/flask/spouty
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]
